### PR TITLE
Handle tests marked as inconclusive

### LINKED ISSUE #2400: Bring back 'InconclusiveCount' property (or alternative) on returned Pester.Run object
### Checklist

- [X] Feature request has a meaningful title
- [X] I have searched the existing issues. See [all issues](https://github.com/pester/Pester/issues?q=is%3Aissue)
- [X] I have tested using the latest version of Pester. See [Installation and update guide](https://pester.dev/docs/introduction/installation).

### Summary of the feature request

In Pester v4 I have used `Set-ItResult -Skipped` and `Set-ItResult -Inconclusive` in tests, and distinguished between the two when generating reports of the results. In v4 it was easy because InconclusiveCount was returned, but while working on moving to v5, I noticed the property was missing, and found this [breaking changes](https://pester.dev/docs/migrations/breaking-changes-in-v5) note in the docs:

> Inconclusive and Pending states are currently no longer available, -Pending and -Inconclusive are translated to -Skip both on test blocks and when using Set-ItResult

What are the odds of this property - or some alternative means of reliably reporting on "Skipped reason" - to find it's way to the Pester.Run object?

### How should it work?

Surface properties similar to what is available in Pester v4.

Alternatively (might be easier to implement) include some kind of "reason data" in the exception thrown by the Set-ItResult command. One option could be to include the data in the [targetObject Dictionary](https://github.com/pester/Pester/blob/a50354e46344aa8d4b4460e55fd2137939d0a215/src/csharp/Pester/Factory.cs#L46) object that is already included in the exception. In theory that logic could also allow specifying custom reasons with a parameter for the `Set-ItResult` command.

### PR DESCRIPTION
**Pull request by csandfeld**
## PR Summary

- Adds properties to CSharp classes to enable `Inconclusive` and `InconclusiveCount` properties on the `Pester.Run` object.
- Adds logic to group and count tests marked as inconclusive.
- Pester Report output updated to support Inconclusive counts etc.
- Adds deprecation notice shown when `Set-ItResult -Pending` is used in tests.
- NUnit reports updated to handle inconclusive tests (needs thorough review).
- Minor changes to help text

Fix #2400

## PR Checklist

- [x] PR has meaningful title
- [x] Summary describes changes
- [x] PR is ready to be merged
  - If not, use the arrow next to `Create Pull Request` to mark it as a draft. PR can be marked `Ready for review` when it's ready.
- [ ] Tests are added/update *(if required)*
- [ ] Documentation is updated/added *(if required)*

## Notes

I think the logic to handle inconclusive tests is sound, but I am looking for feedback on these points

- Are implemented changes to NUnit reports sufficient, or did I miss anything?
- Does this PR call for new tests or changes to existing ones?
- Are documentation updates needed (if so, where)?
- Deprecation notice wording and placement

### CODE REVIEW

File: src/Pester.Runtime.ps1
Comment (fflaten) [inline src/Pester.Runtime.ps1:687]: Wrong for inconclusive, but mostly used for flow tracing, so we can leave it for now.

File: src/Pester.Runtime.ps1
Comment (fflaten) [inline src/Pester.Runtime.ps1:2276]:
```suggestion
                    $b.InconclusiveCount += $child.InconclusiveCount
```

Typo. Would break `InconclusiveCount` in parent blocks. See related task in review comment.

**Review by fflaten — CHANGES_REQUESTED** [review]
Happy new year! 🎆 
Thanks for looking into this. It looks great!

I've left a few comments. There's also a few more places to update:

- Update [example in `Set-ItResult` comment-based help](https://github.com/pester/Pester/blob/f5626bfcd470e677ca53c397fa2fdc1d0070d49e/src/functions/Set-ItResult.ps1#L29-L44) to include a Inconclusive test + fix summary line which has been wrong the whole time. 🤦‍♂️  

- Update `PostPorcess-ExecutedBlock` [here](https://github.com/pester/Pester/blob/f5626bfcd470e677ca53c397fa2fdc1d0070d49e/src/Pester.Runtime.ps1#L2214-L2216) to check and increase `OwnInconlusiveCount` similar to skipped. This is used by `Inconclusive` and `OwnInconclusive` at Block-level in the result-object (`$result.Inconclusive.Block | ft Name, OwnInconclusiveCount, InconclusiveCount`). I've suggested a typo fix to related bug in the review.

- We need some tests to make sure this works and doesn't break in the future. Suggestions:
  - Add test for behavior and error [here](https://github.com/pester/Pester/blob/main/tst/functions/Set-ItResult.Tests.ps1)
  - Extend [this test](https://github.com/pester/Pester/blob/acc66a965219a8e70981977289227a211af4c70e/tst/Pester.RSpec.ResultObject.ts.ps1#L73C57-L73C57) to have a inconclusive test and verify that `Inconclusive` and `InconclusiveCount` properties in the result object are correct
  - Add or extend test to make sure inconclusive attribute works as expected in exported [NUnit2.5](https://github.com/pester/Pester/blob/main/tst/Pester.RSpec.TestResults.NUnit25.ts.ps1) and [NUnit3](https://github.com/pester/Pester/blob/main/tst/Pester.RSpec.TestResults.NUnit3.ts.ps1) reports

- Include `InconclusiveCount` in `ConvertTo-Pester4Result` [here](https://github.com/pester/Pester/blob/f5626bfcd470e677ca53c397fa2fdc1d0070d49e/src/Main.ps1#L1450-L1466)

File: src/functions/Output.ps1
Comment (csandfeld) [reply src/functions/Output.ps1:316]: I agree, and think it needs a blank line as well to not clutter the "Passed/Failed/Skipped" line. How do you like this?

```PowerShell
Pester v5.5.0

Starting discovery in 1 files.
Discovery found 5 tests in 8ms.
Running tests.
Describing a
  [+] a 6ms (3ms|3ms)
  [-] b 5ms (3ms|1ms)
   Expected $false, but got $true.
   at $true | Should -BeFalse, :10
   at <ScriptBlock>, <No file>:10
  [?] c is inconclusive, because INCONCLUSIVE: demo inconclusive 4ms (2ms|2ms)
  [!] d is pending, because PENDING: demo pending 5ms (3ms|2ms)
  [!] e is skipped, because demo skipped 4ms (2ms|1ms)
Tests completed in 82ms
Tests Passed: 1, Failed: 1, Skipped: 2, Inconclusive: 1, NotRun: 0

**DEPRECATED**: The Set-ItResult -Pending parameter is deprecated. It will be removed in a future version of Pester.
```

**Comment by csandfeld** [comment]
> Happy new year! 🎆 Thanks for looking into this. It looks great!

Thank you @fflaten and late happy new year to you too (sorry about the response time).

I have tried to address your comments and requested changes, but would appreciate some advise on this one though:

> * We need some tests to make sure this works and doesn't break in the future. Suggestions:
>   
>   * Add test for behavior and error [here](https://github.com/pester/Pester/blob/main/tst/functions/Set-ItResult.Tests.ps1)

Can I trouble you for an example of what you have in mind that is not already there?

**Comment by csandfeld** [comment]
@fflaten when you have a moment, please let me know if there is anything else I should do or address to complete this PR.

Thanks

**Review by fflaten — CHANGES_REQUESTED** [review]
Getting close! A few more things for the NUnit-reports:
- Testsuites in NUnit 2.5 will get Success result when it contains a inconclusive test. Need to extend https://github.com/pester/Pester/blob/acc66a965219a8e70981977289227a211af4c70e/src/functions/TestResults.NUnit25.ps1#L395-L403
- Results in NUnit 3.0 should be `Inconclusive` if no other tests are failed/passed. Maybe add a `PassedCount -gt 0` = Passed elseif and default to `Inconclusive`?: https://github.com/pester/Pester/blob/acc66a965219a8e70981977289227a211af4c70e/src/functions/TestResults.NUnit3.ps1#L344-L349

**Comment by fflaten** [comment]
> @fflaten when you have a moment, please let me know if there is anything else I should do or address to complete this PR.

Apologies for the delay. Been occupied with work for a while.
We're also waiting on @nohwnd for final review + merge & release. 

> > * We need some tests to make sure this works and doesn't break in the future. Suggestions:
> >   
> >   * Add test for behavior and error [here](https://github.com/pester/Pester/blob/main/tst/functions/Set-ItResult.Tests.ps1)
> 
> Can I trouble you for an example of what you have in mind that is not already there?

Can't remember tbh. so just ignore it. I probably missed the existing inconclusive test 🙂

**Comment by nohwnd** [comment]
& release. <- Not so easy :D My certificate expired because I asked for renew and then went on vacation. I am waiting for a new one.

File: src/functions/Set-ItResult.ps1
Comment (nohwnd) [reply src/functions/Set-ItResult.ps1:68]: Removed.

File: src/functions/Output.ps1
Comment (nohwnd) [reply src/functions/Output.ps1]: Slightly reworded the deprecation message, to point out that it is the parameter and not Set-ItResult that is being deprecated.

**Comment by nohwnd** [comment]
Looking good. I've added cheaper way of checking that the deprecation should be shown so we don't have to inspect every single test on every run.

**Comment by nohwnd** [comment]
Merged, thank you! :)

**Comment by csandfeld** [comment]
Thank you for accepting the PR @nohwnd. And thank you @fflaten for you patience with me through the code reviews.

**Comment by nohwnd** [comment]
No thank you for your work, and sorry for being slow! Same huge thanks for Frode 👏

**Pull request merged**
